api/app/src/main/kotlin/packit/model/dto/UpdateRolePermission.kt

```diff
@@ -7,12 +7,22 @@ data class UpdateRolePermissions(
     val removePermissions: List<UpdateRolePermission> = listOf()
 )
 
-data class UpdatePacketReadRoles(
-    val packetId: String? = null,
+data class UpdateReadRoles(
     val roleNamesToAdd: Set<String> = setOf(),
     val roleNamesToRemove: Set<String> = setOf()
 )
 
+/**
+ * @param cantRead Roles and users that are explicitly restricted from reading the resource.
```

Comment:
> Roles and users that are **explicitly** restricted from reading the resource.

I don't think "explicitly" is the right word here. That would imply that they got blacklisted or banned from the resource, which is not the case.

Maybe more like "roles and users which have no permission to read the resource, directly or indirectly".

Can you clarify how these lists relate to one another? Are they mutually exclusive? Does withRead imply canRead?